agent.go

```diff
@@ -296,6 +296,7 @@ func (a *agent) executeTools(ctx context.Context, message *Message) (*Message, e
 	for i, part := range message.Parts {
 		switch v := any(part).(type) {
 		case ToolPart:
+			i, v := i, v
```

Comment:
In Go 1.22, the official release has already fixed this closure issue, and our go.mod minimum version requirement is Go 1.24. (https://go.dev/doc/go1.22#language)